### PR TITLE
Issue27

### LINKED ISSUE #27: Add the ability to filter by skill type when performing at GET on skills
For example:
GET    /skills?skilltype=scripted
would only return skills with the scripted skill type

### PR DESCRIPTION
**Pull request by Hopding**
Fixes #27

### CODE REVIEW

**Review by thebho — CHANGES_REQUESTED** [review]
Add some tests to prove the filtering works

File: handler/skill_test.go [hunk 1]
```diff
@@ -151,7 +157,40 @@ func TestPerformGetError(t *testing.T) {
...
+}
+
+func TestGetSkillsFiltered(t *testing.T) {
+	skillsConnector = data.NewAccessor(MockDataAccessor{})
+	r := httptest.NewRequest(http.MethodGet, "/skills?skilltype=scripted", nil)
+
+	w := httptest.NewRecorder()
+	err := performGet(w, r)
+	if(err != nil) {
+		fmt.Errorf("Did not expect error when getting skills with filter")
```
Comment (thebho) [inline handler/skill_test.go]: These should all be t.Errorf() then testing.  fmt.Errorf() creates a new error object with formatted text, but doesn't return it to the testing tool.

File: handler/skills.go [hunk 2]
```diff
@@ -127,3 +178,40 @@ func checkForId(url *url.URL) string {
 	}
 	return ""
 }
+
+/*
+extractSkillFilter() returns the filter value of the "skilltype" filter within
```
Comment (thebho) [inline handler/skills.go]: Again, I think this is probably overkill for comments, especially on a non-exported function.  I know by the function name what this does, and the logic and error messages are self explanatory.

**Review by thebho — CHANGES_REQUESTED** [review]

File: handler/skill_test.go (hunk 1 above)
Comment (Hopding) [reply handler/skill_test.go]: Fixed

File: handler/skills.go (hunk 2 above)
Comment (Hopding) [reply handler/skills.go]: I've removed the comments for `extractSkillFilter()`. Should be ready now 👍

**Comment by thebho** [comment]
Looks good.  Does this work with the new Controllers in master?

**Comment by Hopding** [comment]
It does now 😺

**Review by thebho — APPROVED** [review]

**Pull request merged**